### ISSUE TITLE
urn:ietf:params:acme:error:unauthorized :: Incorrect TXT record

### ISSUE THREAD

**Issue by philthynz** [vancluever/terraform-provider-acme#190]
I have the following config:

**provider.tf**
```
terraform {
  required_providers {
    acme = {
      source  = "vancluever/acme"
      version = "~> 2.5.0"
    }
  }
}

# Use acme provider prod
provider "acme" {
  alias      = "prod"
  server_url = "https://acme-v02.api.letsencrypt.org/directory"
}

# Use acme provider staging
provider "acme" {
  alias      = "staging"
  server_url = "https://acme-staging-v02.api.letsencrypt.org/directory"
}
````

**main.tf**
```
# ACME registration needs a private key
resource "tls_private_key" "private_key" {
  algorithm = "RSA"
}

# ACME registration
resource "acme_registration" "reg" {
  account_key_pem = tls_private_key.private_key.private_key_pem
  email_address   = var.EMAIL
  provider        = acme.staging
}

# Random password for certificate
resource "random_password" "password" {
  length           = 16
  special          = true
  override_special = "_%@"
}

# Generate certificate
resource "acme_certificate" "certificate" {
  account_key_pem           = acme_registration.reg.account_key_pem
  common_name               = var.DOMAIN_NAME
  provider                  = acme.staging
  subject_alternative_names = var.SANS
  min_days_remaining        = var.MIN_DAYS_REMAINING
  certificate_p12_password  = random_password.password.result
  key_type                  = var.KEY_TYPE

  dns_challenge {
    provider = "azure"
    config = {
      ARM_SUBSCRIPTION_ID   = var.AZURE_SUBSCRIPTION_ID
      AZURE_CLIENT_ID       = var.AZURE_CLIENT_ID
      AZURE_CLIENT_SECRET   = var.AZURE_CLIENT_SECRET
      AZURE_ENVIRONMENT     = "public"
      AZURE_SUBSCRIPTION_ID = var.AZURE_SUBSCRIPTION_ID
      AZURE_TENANT_ID       = var.AZURE_TENANT_ID
      AZURE_RESOURCE_GROUP  = var.DNS_RESOURCE_GROUP
    }
  }
}
```

And pass in the required values as needed:

```
Set-Item -Path env:TF_VAR_DOMAIN_NAME -Value "domain.tld.com"
Set-Item -Path env:TF_VAR_EMAIL -Value "myemail"
Set-Item -Path env:TF_VAR_DNS_RESOURCE_GROUP -Value "res-grp"
Set-Item -Path env:TF_VAR_AZURE_SUBSCRIPTION_ID -Value "subid"
Set-Item -Path env:TF_VAR_KEY_TYPE -Value "2048"
Set-Item -Path env:TF_VAR_AZURE_CLIENT_ID -Value "clientid"
Set-Item -Path env:TF_VAR_AZURE_CLIENT_SECRET -Value 'secret'
Set-Item -Path env:TF_VAR_AZURE_TENANT_ID -Value "tenantid"
Set-Item -Path env:TF_VAR_MIN_DAYS_REMAINING -Value "30"
```

A few days ago I could generate certs for this single domain fine, and now it has an error:

```
urn:ietf:params:acme:error:unauthorized :: Incorrect TXT record "pT0exxxxxxxdXo" (and 1 more) found at _acme-challenge.domain.tld.com
```

I have tried the same request with acme.sh and don't have the same issues:

```
export AZUREDNS_SUBSCRIPTIONID="subid"
export AZUREDNS_TENANTID="tenantid"
export AZUREDNS_APPID="clientid"
export AZUREDNS_CLIENTSECRET='secret'

acme.sh --staging --issue --dns dns_azure -d domain.tld.com
```

Has something changed in the terraform provider to break this?

**Comment by philthynz** [comment]
Strangely enough. I can still use the same config to generate certs for new dns names, but it doesn't work for existing one's where there was a previous cert. Usually the certs are stored in a TF statelock, and I deleted the statelock to force a new cert.

**Comment by vancluever** [comment]
@philthynz by any chance are there broken stale TXT records on those TLDs?

**Comment by philthynz** [comment]
Yes I believe that is the issue, but I have waited a few days and their are still old records in the DNS:

![image](https://user-images.githubusercontent.com/17504095/124519180-aa35f900-de3c-11eb-919b-6ca4ca2e013f.png)

That's using 8.8.8.8.

I have placed a record with value "hello", and 1.1.1.1 seems to see it fine:

![image](https://user-images.githubusercontent.com/17504095/124519268-e5d0c300-de3c-11eb-80f5-d1ce45d000e9.png)

So the issue seems to be most other DNS servers have not updated, even after 4 days or so....

I've tried with `recursive_nameservers     = ["1.1.1.1:53"]`, and that still doesn't seem to work.

Would there be a way I can see what name server the provider is using in a log somewhere?
